sync/class.jetpack-sync-wp-replicastore.php

```diff
@@ -745,18 +750,32 @@ private function table_checksum( $table, $columns, $id_column, $where_sql = '1=1
 			$columns_sql = implode( ',', $sanitized_columns );
 		}
 
-		if ( $min_id !== null ) {
-			$min_id     = intval( $min_id );
-			$where_sql .= " AND $id_column >= $min_id";
-		}
+		if ( $min_id !== null && $max_id !== null ) {
+			if ( $min_id == $max_id ) {
+				$min_id = intval( $min_id );
+				$where_sql = " $id_column = $min_id LIMIT 1";
+			} else {
+				if ( $min_id !== null ) {
```

Comment:
This `if` shouldn't be necessary - we're already certain that this is true, because we checked in the parent `if`: https://github.com/Automattic/jetpack/pull/12151/files#diff-83da856148ee51e9120cfa9c044afa5cR753

Reply:
c8533fc4a